distributed/scheduler.py

```diff
@@ -2857,23 +2857,21 @@ def check_idle_saturated(self, ws: WorkerState, occ: float = -1.0) -> None:
 
         p = len(ws.processing)
 
-        idle = self.idle
-        saturated = self.saturated
-        saturated.discard(ws)
-        if self.is_unoccupied(ws, occ, p):
-            if ws.status == Status.running:
-                idle[ws.address] = ws
+        self.saturated.discard(ws)
+        if ws.status != Status.running:
+            self.idle.pop(ws.address, None)
```

Comment:
Defensive programming. This should not actually be needed.

Reply:
How about popping this every time? Looks like this is always popped unless we're adding it explicitly, i.e. we're touching the dict either way. This way we can apply this defensive programming w/out adding another if clause.